cmake/FindCMSIS.cmake

```diff
@@ -156,8 +156,20 @@ IF(STM32_CHIP_TYPE)
         CMAKE_FIND_ROOT_PATH_BOTH
     )
     LIST(APPEND CMSIS_SOURCES ${CMSIS_STARTUP_SOURCE_FILE})
+
+    SET(CMSIS_COMPILE_DEFINITIONS "STM32F${STM32_CHIP_TYPE}")
 ENDIF()
 
 INCLUDE(FindPackageHandleStandardArgs)
 
 FIND_PACKAGE_HANDLE_STANDARD_ARGS(CMSIS DEFAULT_MSG CMSIS_INCLUDE_DIRS CMSIS_SOURCES)
+
+IF(CMSIS_FOUND AND NOT TARGET STM32Cube::CMSIS)
+    ADD_LIBRARY(STM32Cube::CMSIS INTERFACE IMPORTED)
+    SET_TARGET_PROPERTIES(STM32Cube::CMSIS 
```

Comment:
Why Don't you use ?
```
target_include_directories()
target_sources()
target_compile_definitions()
```

Reply:
TL;DR; You're right. 

At the time of this PR, I just followed kind of blindly the article that I mentioned in https://github.com/ObKo/stm32-cmake/pull/87#issuecomment-448932193.

The fact is that when one declares `IMPORTED` targets, functions like `target_*()` can't be used and we have to modify the target using the low-level `set_target_properties()` function.
But actually in this case AFAIK, since the target that we're declaring is INTERFACE, using `IMPORTED` or not doesn't matter much. `IMPORTED` targets should really be used when we have a library that is already built somewhere.

As of now, I think we should remove the `IMPORTED` specifier and declare the targets as `STATIC`, for the following reason: when declaring some target as INTERFACE we're just saying that everything that links to that target has to take its properties too; specifically in our case, where we have INTERFACE targets that include a series of sources, all the targets that link to these INTERFACE targets will compile those source files. While this might be useful when we deal with targets like STM32HAL (which requires a `stm32fxhal_conf.h` to lay somewhere) and generally with targets that are not self-contained, since everything the INTERFACE target specifies is compiled together with the target that links it, allowing us to define the "missing pieces" of the INTERFACE target in the target that links it, the INTERFACE target stuff breaks really quickly once the build system starts getting a little complex.

Imagine when we want to link the STM32HAL target to multiple executables. Or if we have our project organized into modules and each module should be as self-contained as possible, i.e. should specify all its dependencies. Using the INTERFACE target method, for every target that links the INTERFACE target, the sources files specified for the INTERFACE target are re-compiled, which is really bad.
Using the STATIC target method, we have instead that the target is built stand-alone and the produced *.a library file is linked to every target that links it.
The only drawback of STATIC targets is that if they're not really stand-alone, well... they can't be built stand-alone... Which is indeed the case for STM32HAL, which requires the presence of `stm32xx_hal_conf.h` to complete the puzzle.

Currently, in order to solve this problem, I require the user to define a target (e.g. for the case of STM32HAL, let's say that I require a target called `STM32HALConf`, which must specify the include directory where `stm32xx_hal_conf.h` can be found) before calling `find_package`, and then the target that I define in the find module will link the target provided by the user.

Let's just say that, while the INTERFACE target method works by telling the user "I give the incomplete puzzle, do as you want in the application layer", the STATIC target method works by telling the user "Give me this missing piece and I'll give you the nice, self-contained target that you just need to blindly link".